src/components/vanilla/charts/KPIChart/index.tsx

```diff
@@ -1,34 +1,39 @@
-import { DataResponse } from '@embeddable.com/core';
+import { DataResponse, TimeRange } from '@embeddable.com/core';
 import React, { useMemo } from 'react';
 
-import format from '../../../util/format';
+import formatValue from '../../../util/format';
 import Container from '../../Container';
 import { WarningIcon } from '../../icons';
 
 type Props = {
   value: DataResponse;
+  prevValue?: DataResponse;
+  prevTimeFilter?: TimeRange;
   title: string;
   prefix: string;
   suffix: string;
   metric: { name: string; meta?: object };
+  dps: number;
 };
 
 export default (props: Props) => {
-  const n = useMemo(() => {
-    if (!props.value?.data?.length) return;
+  const { n, percentage } = useMemo(() => {
+    if (!props.value?.data?.length) return { percentage: null };
 
-    if (!props.metric?.name) return;
+    if (!props.metric?.name) return { percentage: null };
 
-    const n = props.value.data[0][props.metric?.name];
+    const n = parseFloat(props.value.data[0][props.metric?.name]);
 
-    const num = parseFloat(n);
+    const prev = parseFloat(props.prevValue?.data?.[0]?.[props.metric?.name]);
```

Comment:
improvement: this lines can be merged into 1. 

`if(!props.value?.data?.length || !props.metric?.name) return { percentage: null };`

Reply:
@mad-raccoon fair point. Done